src/QsFmt/Formatter/ParseTree/Statement.fs

```diff
@@ -104,51 +104,85 @@ type StatementVisitor(tokens) =
+            Keyword = context.``return`` |> Node.toTerminal tokens
             Expression = expressionVisitor.Visit context.value
             Semicolon = context.semicolon |> Node.toTerminal tokens
         }
         |> Return
 
-    override visitor.VisitQubitDeclaration context =
+    override _.VisitFailStatement context =
         {
-            Kind =
-                match context.keyword.Text with
-                | "use"
-                | "using" -> Use
-                | _ -> Borrow
-            Keyword = context.keyword |> Node.toTerminal tokens
-            OpenParen = context.openParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
-            Binding = context.binding |> qubitBindingVisitor.Visit
-            CloseParen = context.closeParen |> Option.ofObj |> Option.map (Node.toTerminal tokens)
-            Coda =
-                if context.body <> null then
-                    {
-                        OpenBrace = context.body.openBrace |> Node.toTerminal tokens
-                        Items = context.body._statements |> Seq.map visitor.Visit |> List.ofSeq
-                        CloseBrace = context.body.closeBrace |> Node.toTerminal tokens
-                    }
-                    |> Block
-                else
-                    context.semicolon |> Node.toTerminal tokens |> Semicolon
+            Keyword = context.fail |> Node.toTerminal tokens
+            Expression = expressionVisitor.Visit context.value
+            Semicolon = context.semicolon |> Node.toTerminal tokens
         }
-        |> QubitDeclaration
+        |> Fail
 
     override _.VisitLetStatement context =
         {
-            LetKeyword = context.``let`` |> Node.toTerminal tokens
+            Keyword = context.``let`` |> Node.toTerminal tokens
             Binding = symbolBindingVisitor.Visit context.binding
             Equals = context.equals |> Node.toTerminal tokens
             Value = expressionVisitor.Visit context.value
             Semicolon = context.semicolon |> Node.toTerminal tokens
         }
         |> Let
 
+    override _.VisitMutableStatement context =
+        {
+            Keyword = context.``mutable`` |> Node.toTerminal tokens
+            Binding = symbolBindingVisitor.Visit context.binding
+            Equals = context.equals |> Node.toTerminal tokens
+            Value = expressionVisitor.Visit context.value
+            Semicolon = context.semicolon |> Node.toTerminal tokens
+        }
+        |> Mutable
+
+    override _.VisitSetStatement context =
+        {
+            Keyword = context.set |> Node.toTerminal tokens
+            Binding = symbolBindingVisitor.Visit context.binding
+            Equals = context.equals |> Node.toTerminal tokens
+            Value = expressionVisitor.Visit context.value
+            Semicolon = context.semicolon |> Node.toTerminal tokens
+        }
+        |> SetStatement
+
+    override _.VisitSetUpdateStatement context =
```

Comment:
Why the discrepancy in naming between the parse tree and the rewriter, the reducer and the syntax tree?

Reply:
Hopefully I've resolved the discrepancies here.